web/js/MapUI/Components/MouseMoveEvents/MouseMoveEvents.js

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {
+  throttle as lodashThrottle,
+} from 'lodash';
+import util from '../../../util/util';
+import {
+  MAP_MOUSE_MOVE,
+  MAP_MOUSE_OUT,
+} from '../../../util/constants';
+
+const { events } = util;
+
+const MouseMoveEvents = (props) => {
+  const {
+    isCoordinateSearchActive,
+    isEventsTabActive,
+    isMobile,
+    isMeasureActive,
+    isMapAnimating,
+    sidebarActiveTab,
+    map,
+    ui,
+    compareMapUi,
+  } = props;
+
+  const [mouseMove, setMouseMove] = useState({});
+
+  const throttledOnMouseMove = lodashThrottle(({ pixel }) => {
+    if (!map.ui.selected) return;
+
+    const coords = ui.selected.getCoordinateFromPixel(pixel);
+
+    if (map.ui.selected.proj !== ui.selected.proj) return;
+    if (ui.mapIsbeingZoomed) return;
+    if (ui.mapIsbeingDragged) return;
+    if (compareMapUi && compareMapUi.dragging) return;
+    if (isMobile) return;
+    if (isMeasureActive) return;
+    if (isCoordinateSearchActive) return;
+    if (!coords) return;
+    if (isEventsTabActive || isMapAnimating || sidebarActiveTab === 'download') return;
+    ui.runningdata.newPoint(pixel, ui.selected);
+  }, 300);
+
+  useEffect(() => {
+    throttledOnMouseMove(mouseMove);
+  }, [mouseMove]);
+
+  events.on(MAP_MOUSE_MOVE, setMouseMove);
+  events.on(MAP_MOUSE_OUT, (e) => {
+    throttledOnMouseMove.cancel();
+    ui.runningdata.clearAll();
+  });
+
+  return null;
+};
+
+const mapStateToProps = (state) => {
+  const {
+    events, locationSearch, sidebar, animation, measure, screenSize, map,
```

Comment:
"events" is already declared on line 13.

Reply:
The events called here in `mapStateToProps` represents our Redux global state object for our natural events data while the events variable declared above the component on line 13 represents all of the event listeners from our main `util.js` file. The reason that this isn't a problem is because I am only destructing events in mapStateToProps from the `state` object and then using the property `events.active`. So since I am not returning `events` in `mapStateToProps` (which I would have to list in `props` in the component) it does not cause an issue.